### DISCUSSION TITLE
Best guide to use for a mosaic??

### DISCUSSION THREAD

**Discussion by mconsidine** [quatrope/astroalign#95] [Help Desk]
Apologies in advance if I have overlooked something obvious.  This isn't really an issue, per se

I am trying to create a mosaic of two images, where the resulting image is the combination of the two original images, but without any cropping.

I can get the registered image from the examples I've seen, but I can't seem to get the resulting combined image.

Have I overlooked a parameter that turns off the cropping so that the result has dimensions covering the original as well as rotated/transformed image?  I got the test_align.py script to execute without error.  If there is something in there I should build upon, I'd appreciate a pointer.

TIA,
mconsidine

**Comment by martinberoiz** [comment]
Hi,

To create a mosaic, you can pad the target image with a pad region large enough to contain the transformed image entirely. Then before you call the `register` method you mask the original target region as explained here:

https://astroalign.quatrope.org/en/latest/mask.html#selecting-regions-with-masking

I think that should work. Let me know if you need more help.

**Comment by mconsidine** [comment]
Thanks for that.  I'm working with two color fits images.  So after selecting one channel out of each and adjust the histogram to knock down the number of stars, I get a transform that seems to work.  Except that I really need to extract that transform, apply it to the original source image and then coadd the two I think.

The resulting 'registered' file, when written to first, is very large. Is there a bounding box function available that would help trim off what is unnecessary? 

Thanks again for your help and for the package.
mconsidine

**Comment by martinberoiz** [comment]
Astroalign itself does not have any tool to downsample or trim images. But you can do an easy trim using [numpy array slicing](https://numpy.org/doc/stable/user/basics.indexing.html#slicing-and-striding) on the registered ndarray or [astropy Cutout2D](https://docs.astropy.org/en/stable/api/astropy.nddata.utils.Cutout2D.html) if you need to maintain the WCS info as well.

I'm not sure if this answer your question.

**Comment by mconsidine** [comment]
It does answer my question - thank you.  And I seem to generally have things working out, except that I can't quite get the masking right to include both the target and the source.  The source is correctly positioned inside a larger padded image, but the target image hasn't come through.  If you have a further example you can point me to, I'll pull this apart.  I'm very appreciative of the help you've given already.
Rgds,
mconsidine

**Reply by martinberoiz** [reply]
I converted this issue into a "discussion".

If you send me some working code along with a couple of images I can try helping you create the mosaic. Otherwise it's hard to give you a concrete example. There's a lot of finicking sometimes with images.

**Comment by mconsidine** [comment]
Thank you for switching this to "discussion" as I should have started out with that.I can send working code and links to 2 images tomorrow.I'm obliged.MattOn Feb 22, 2024 7:01 PM, Martin Beroiz ***@***.***> wrote:
I converted this issue into a "discussion".
If you send me some working code along with a couple of images I can try helping you create the mosaic. Otherwise it's hard to give you a concrete example. There's a lot of finicking sometimes with images.

**Reply by mconsidine** [reply]
Thank you again for your offer of help/tips.  I've sent you links to 2 files and the code I'm using is attached here.  If the two files A and B are in the same directory as the code, this should run "as-is."  I've tried too to comment what I was doing/thinking at each point.
[astroalign_test2.txt](https://github.com/quatrope/astroalign/files/14389572/astroalign_test2.txt)

**Reply by mconsidine** [reply]
Hello again,

I'm continuing to work on this and saw that the "mask" tutorial might be something I could use.  The text doesn't show how the final result was obtained so I have been trying to replicate that on my own.

Would it be possible for you to present (or point me to, if I missed it) the workflow that place the source_half_degree.png file into the target image?  I seem to be running into a matching triangles issue and am not sure what I need to tweak to reproduce those results.
Many thanks,
mconsidine

**Reply by martinberoiz** [reply]
Hi, apologies for not replying earlier. I just sent you an email with a jupyter notebook where you can see how you can make a mosaic with the images.

The final mosaic does not look well because of gain differences in the images (or something related to that, I didn't look) but the idea is there.

Let me know if it helps.
